### PR TITLE
Salmon test and modules

### PR DESCRIPTION
**Pull request by J35P312**
Hello there!
I have made test and modules for the salmon software. Salmon is a tool for transcript quantification:

https://combine-lab.github.io/salmon/

### CODE REVIEW

File: lib/MIP/Program/Variantcalling/Salmon.pm [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+## Constants
+Readonly my $SPACE => q{ };
+
+sub salmon_index {
+
+## Function  : Perl wrapper for Salmon index, version 0.9.1.
+## Returns   : @commands
+## Arguments : $fasta_path             => Input reference fasta path, note salmon does not use the genome reference fasta, it uses a fasta file of transcripts
+##           : $FILEHANDLE             => Filehandle to write to
+##           : $outfile_path           => outfile path
```
Comment (henrikstranneheim) [inline lib/MIP/Program/Variantcalling/Salmon.pm]: outfile -> Outfile

File: lib/MIP/Program/Variantcalling/Salmon.pm [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+            strict_type => 1,
+        },
+    };
+
+    check( $tmpl, $arg_href, 1 ) or croak q{Could not parse arguments!};
+
+    ## Stores commands depending on input parameters
+    my @commands = q{salmon index};
+
+    #options
```
Comment (henrikstranneheim) [inline lib/MIP/Program/Variantcalling/Salmon.pm]: #options -> # Options

File: lib/MIP/Program/Variantcalling/Salmon.pm [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+## Returns   : @commands
+## Arguments : $FILEHANDLE                 => Filehandle to write to
+##           : $index_path                 => Path to the index folder
+##           : $lib                        => Library visit the salmon website for more  info
+##           : $outfile_path               => The path of the  output directory
+##           : $read_1_fastq_path          => Read 1 Fastq path
+##           : $read_2_fastq_path          => Read 2 Fastq path
+##           : $read_files_command         => command applied to the input FASTQ files
+##           : $stderrfile_path            => Stderrfile path
+##           : $stderrfile_path_append     => Append stderr info to file path
```
Comment (henrikstranneheim) [inline lib/MIP/Program/Variantcalling/Salmon.pm]: use single space between txt and fat comma (=>)

File: lib/MIP/Program/Variantcalling/Salmon.pm [hunk 4]
```diff
@@ -0,0 +1,249 @@
...
+            required    => 1,
+            store       => \$read_1_fastq_path,
+            strict_type => 1,
+        },
+        read_2_fastq_path => {
+            defined     => 1,
+            store       => \$read_2_fastq_path,
+            strict_type => 1,
+        },
+
```
Comment (henrikstranneheim) [inline lib/MIP/Program/Variantcalling/Salmon.pm]: Remove newline

File: lib/MIP/Program/Variantcalling/Salmon.pm [hunk 5]
```diff
@@ -0,0 +1,249 @@
...
+    my @commands = q{salmon quant};
+
+    push @commands, q{--index} . $SPACE . $index_path;
+
+    push @commands, q{--libType} . $SPACE . $lib_type;
+
+    push @commands, q{--output} . $SPACE . $outfile_path;
+
+    push @commands,
+        q{-1}
```
Comment (henrikstranneheim) [inline lib/MIP/Program/Variantcalling/Salmon.pm]: What is this? Would be nice with a comment

File: lib/MIP/Program/Variantcalling/Salmon.pm [hunk 6]
```diff
@@ -0,0 +1,249 @@
...
+
+    ## Stores commands depending on input parameters
+    my @commands = q{salmon quant};
+
+    push @commands, q{--index} . $SPACE . $index_path;
+
+    push @commands, q{--libType} . $SPACE . $lib_type;
+
+    push @commands, q{--output} . $SPACE . $outfile_path;
+
```
Comment (henrikstranneheim) [inline lib/MIP/Program/Variantcalling/Salmon.pm]: What is this? Would be nice with a comment for the overall function

**Comment by J35P312** [comment]
I did some cleaning and added comments. Do the comments make sense?
Good luck =P

**Comment by henrikstranneheim** [comment]
👍

**Pull request merged**